### PR TITLE
feat(#issue37): implements `mapstruct` tag

### LINKED ISSUE #37: Feat: support recursive masking for map values via mask:"map_struct"
# Expected behaviour

`m.Struct` should support map value masking when field tag is `mask:"map_struct"`.

For map fields with `mask:"map_struct"`:
- value type `struct` should be recursively masked
- value type `*struct` should be recursively masked
- nil map should not panic
- nil pointer value in map should be kept as nil

For map fields without `mask:"map_struct"`:
- value should remain unchanged

# Actual behaviour

Before this change, map fields were not consistently handled for recursive masking in `m.Struct`.

# Use case

I store sensitive payloads in map values and want to use `m.Struct` as a generic sanitizer before logging/response.

Example scenarios:
- `map[int]Item`
- `map[int]*Item`

# Reproduction code

```go
package masker

import "testing"

func TestRepro_MapStructMasking(t *testing.T) {
  type Item struct {
    ID string `mask:"id"`
  }

  type Payload struct {
    Items map[int]Item  `mask:"map_struct"`
    Ptrs  map[int]*Item `mask:"map_struct"`
  }

  m := NewMaskerMarshaler()
  masked, err := m.Struct(Payload{
    Items: map[int]Item{1: {ID: "A123456789"}},
    Ptrs:  map[int]*Item{1: {ID: "A123456789"}, 2: nil},
  })
  if err != nil {
    t.Fatalf("unexpected error: %v", err)
  }

  _ = masked
  // expected:
  // Items[1].ID == "A12345****"
  // Ptrs[1].ID  == "A12345****"
  // Ptrs[2]     == nil
}
```

### PR DESCRIPTION
**Pull request by XiXiangFiles**
## Summary
Add support for recursive masking on map values with `mask:"map_struct"`.

## What Changed
- Added new tag type: `map_struct`
- Updated `Struct()` map handling to:
  - recursively mask map values for `struct` and `*struct`
  - keep nil maps and nil pointer values safe
  - skip recursive masking when tag is not `map_struct`
- Added/updated tests for map masking scenarios (table-driven)
- Updated README with `map_struct` docs and examples

## Validation
- `go test ./... -run TestMaskerMarshaler_MapStructs -v`
image

- `go test ./...`
image

## Compatibility
Backward compatible. New behavior is only applied when `mask:"map_struct"` is used.

## Related
Closes #37

### CODE REVIEW

File: masker.go [hunk 1]
```diff
@@ -12,19 +12,20 @@ type MaskerType string
...
+	MaskerTypeEmail     MaskerType = "email"
+	MaskerTypeMobile    MaskerType = "mobile"
+	MaskerTypeTel       MaskerType = "tel"
+	MaskerTypeID        MaskerType = "id"
+	MaskerTypeCredit    MaskerType = "credit"
+	MaskerTypeURL       MaskerType = "url"
+	MaskerTypeAbuse     MaskerType = "abuse"
+	MaskerTypeStruct    MaskerType = "struct"
+	MaskerTypeAll       MaskerType = "all"
+	MaskerTypeMapStruct MaskerType = "map_struct"
```
Comment (ggwhite) [inline masker.go]: **Tag 命名慣例 / Tag naming convention**

現有的 tag 都是單字或縮寫，沒有底線（`none`, `password`, `struct`, `all` 等）。`map_struct` 是唯一帶底線的 tag，打破了命名一致性。

All existing tags use single words or abbreviations without underscores. `map_struct` is the only exception, which breaks the naming convention.

這個 tag 一旦 merge 就是公開 API，之後改會是 breaking change。建議考慮 `mapstruct` 或 `mstruct`。

Since this becomes public API once merged, renaming later would be a breaking change. Consider `mapstruct` or `mstruct` instead.

File: masker.go [hunk 2]
```diff
@@ -310,6 +329,59 @@ func (m *MaskerMarshaler) Struct(s interface{}) (interface{}, error) {
 	return tptr.Interface(), nil
 }
 
+// maskMapStructValue recursively masks map_struct values for struct, ptr, and slice shapes.
+// Unsupported scalar/container leaf values are returned as-is to preserve original data.
```
Comment (ggwhite) [inline masker.go:338]: **建議在 godoc 補充循環引用的限制 / Mention the cycle limitation in the godoc**

README 有說明這個限制，但直接看 function 的人不一定會去讀 README。建議在函式註解補一行：

The README documents this, but users reading the code directly may miss it. Suggest updating the comment:

```go
// maskMapStructValue recursively masks map_struct values for struct, ptr, and slice shapes.
// Unsupported scalar/container leaf values are returned as-is to preserve original data.
// 注意：不支援循環引用的物件圖，遇到時會 stack overflow。
// Note: cyclic object graphs are not handled and will cause a stack overflow.
```

File: masker_test.go [hunk 3]
```diff
@@ -551,3 +551,214 @@ func Test_overlay(t *testing.T) {
...
+			got, err := m.Struct(tt.in)
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("MaskerMarshaler.Struct() = %#v, want %#v", got, tt.want)
+			}
+		})
+	}
+}
```
Comment (ggwhite) [inline masker_test.go:782]: **缺少的邊界測試 / Missing edge case tests**

建議補上以下兩個 case：

1. **純量 map value（如 `map[K]string`）** — 確認 leaf value 原樣通過，不被改動：

```go
{
    name: "map with scalar values passes through unchanged",
    in: struct {
        Items map[string]string `mask:"map_struct"`
    }{Items: map[string]string{"key": "value"}},
    want: &struct {
        Items map[string]string `mask:"map_struct"`
    }{Items: map[string]string{"key": "value"}},
},
```

2. **非 nil 的空 map（len=0）** — 確認空 map 不會出問題：

```go
{
    name: "empty non-nil map",
    in: struct {
        Items map[int]item `mask:"map_struct"`
    }{Items: map[int]item{}},
    want: &struct {
        Items map[int]item `mask:"map_struct"`
    }{Items: map[int]item{}},
},
```

**Review by ggwhite — COMMENTED** [review]
Thanks for the implementation! The core logic looks correct and the test coverage is solid. A few suggestions below.
